src/TableOfContents/index.tsx

```diff
@@ -49,7 +50,10 @@ export type RowComponentProps<T extends TableOfContentsItem, E = {}> = {
   toggleExpanded: () => void;
 } & ({} extends E ? {} : { extra: E });
 
-export type RowComponentType<T extends TableOfContentsItem, E = {}> = React.ComponentType<RowComponentProps<T, E>>;
+export type RowComponentType<T extends TableOfContentsItem, E = {}> = (
+  props: PropsWithChildren<RowComponentProps<T, E>>,
+  context?: any,
+) => ReactElement<any, any> | null;
```

Comment:
This is a subset of the original `React.ComponentType`.

Unforunately `ComponentType` / `FunctionComponent` do not play nice with generics because of `propTypes`. This is the call signature extracted from `FunctionComponent` which is enough for our purposes yet enables generics.

Reply:
Maybe a comment above so no one need to wonder why it is done this way?